src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -46,7 +56,39 @@ class FileIOMicroReadBenchmarkFixture : public MicroBenchmarkBasicFixture {
  protected:
 };
 
-BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, READ_NON_ATOMIC)(benchmark::State& state) {  // open file
+BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, READ_NON_ATOMIC_SEQUENTIAL)(benchmark::State& state) {  // open file
+  int32_t fd;
+  if ((fd = open("file.txt", O_RDONLY)) < 0) {
+    std::cout << "open error " << errno << std::endl;
+  }
```

Comment:
```suggestion
  Assert(fd = open("file.txt", O_RDONLY)) >= 0, "open error: " + strerror(errno));
```

Reply:
Could be done for the other error cases, too.

Reply:
I've tried to implement this, but I got som epermission errors. Can you have a look at my WIP?